pkg/shared/util/grpc_utils.go

```diff
@@ -28,10 +28,47 @@ import (
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/status"
 
+	"github.com/Masterminds/semver/v3"
+
+	"github.com/numaproj/numaflow"
+
 	sdkerr "github.com/numaproj/numaflow/pkg/sdkclient/error"
 	resolver "github.com/numaproj/numaflow/pkg/sdkclient/grpc_resolver"
 )
 
+// isCompatible checks if the current numaflow version is compatible with the given language's SDK version
+func isCompatible(serverInfo *info.ServerInfo) error {
+	type sdkVersions map[info.Language]string
+	var versionMappingConfig = sdkVersions{
+		info.Go:     ">= 0.7.0-0",
+		info.Java:   ">= 0.7.0-0",
+		info.Python: ">= 0.7.0-0",
+	}
```

Comment:
should we have to maintain a matrix?

1.0.0 supports SDK X1, Y1, Z1 versions.
1.1.0 supports SDK X2, Y2, Z2 versions.

Reply:
A matrix would be more cleaner and extensible for sure. 
<img width="593" alt="Screenshot 2024-04-05 at 11 10 41 AM" src="https://github.com/numaproj/numaflow/assets/22337642/64b5ef75-3147-4da6-977a-fbd2a8379886">

Reply:
is there a way that we can dynamically fetch the current version the client is running? Because with the way i have it right now in `version.go`, it would only make sense to embed the supported sdk version for the given numaflow version, i.e all the other rows in the matrix would never be used.

Reply:
Is this the minimum supported version that we are considering or maximum?

Reply:
minimum support version will be the lower bound. upper bound will be what is out there?

Reply:
yes